docs/source/daml/intro/3_Data.rst

```diff
@@ -0,0 +1,202 @@
+2. Most of the variables are declared inside a ``let`` block. That's because the ``scenario do`` block expects scenario actions like ``submit`` or ``getParty``. An integer like ``123`` is not an action, it's a pure expression we can evaluate without any ledger. The ``let`` can be thought of as turning variable declaration into an action.
+3. None of the variables have type annotations. That's because DAML is very good at *inferring* types. The compiler knows that ``123`` is an ``Int`` so if you declare ``my_int = 123``, it can infer that ``my_int`` is also an ``Int`` so you don't have to write ``my_int : Int = 123``. You can always add type annotations to aid readability, and sometimes, in case of ambiguity, have to.
+4. The ``assert`` function is an action that takes a boolean value and succeeds with ``True`` and fails with ``False``. Try putting ``assert False`` somewhere in a scenario and see what happens to the scenario result.
+
+With tempates and these native types, it's already possible to write a schema akin to table in relational database. Below, ``Token`` is extended into a simple ``CashBalance``, administered by a party in the role of an accountant.
+
+.. literalinclude:: daml/3_Data/Native.daml
+  :language: daml
+  :start-after: -- CASH_BALANCE_BEGIN
+  :end-before: -- CASH_BALANCE_END
+
+Assembling types
+----------------
+
+There's quite a lot of information on the ``CashBalance`` above and it would be nice to be able to give that data more structure. Fortunately, DAML's type system has a number of ways to assemble these native types into much more expressive structures.
+
+Tuples
+~~~~~~
+
+A common task is to group values in a generic way. Take, for example a key-value pair with a ``Text`` key and an ``Int`` value. In DAML, you could use a two-tuple of type ``(Text, Int)`` to do so. If you wanted to express a coordinate in three dimensions, you could group three ``Decimal`` values using a three-tuple ``(Decimal, Decimal, Decimal)``.
+
+
+.. literalinclude:: daml/3_Data/Tuple.daml
+  :language: daml
+  :start-after: -- TUPLE_TEST_BEGIN
+  :end-before: -- TUPLE_TEST_END
+
+Note how the data can be accessed either through functions ``fst``, ``snd``, ``fst3``, ``snd3``, ``thd3``, or via a dot-syntax with field names ``_1``, ``_2``, ``_3``, etc.
+
+Lists
+~~~~~
+
+Lists in DAML take a single type parameter defining the type of thing in the list. So you can have a list of integers ``List Int`` or a list of strings ``List Text``, but not a list mixing integers and strings. That's because DAML is statically and strongly typed. When you get an element out of a list, the compiler needs to know what type that element has. The below scenario instantiates a few lists of integers and demonstrates the most important list functions.
+
```

Comment:
```suggestion

Note that, because lists are used fairly often, and to mirror the literal syntax for list values, there is a syntactic shortcut to define a type of list: `[a]` is equivalent to `List a`.

```